services/QuillLMS/deploy.sh

```diff
@@ -71,6 +68,41 @@ case $1 in
     exit 1
 esac
 
+promote() {
+  git fetch origin
+  local latest_prod_release_sha=`heroku releases -a empirical-grammar | grep Deploy | cut -d ' ' -f 4 | head -n 1`
+  local promotion_candidate_sha=`heroku releases -a $HEROKU_APP | grep Deploy | cut -d ' ' -f 4 | head -n 1`
+
+  # if promotion candidate is not a commit descendent of the current production slug, abort
+  git merge-base --is-ancestor $latest_prod_release_sha $promotion_candidate_sha
+
+  if [[ $? != 0 ]]; then
+      echo "Latest deployed ($latest_prod_release_sha) on prod is not an ancestor of promotion candidate ($promotion_candidate_sha) on $HEROKU_APP. Exiting."
+      exit 1
+  fi
+
+  heroku pipelines:promote -a $HEROKU_APP
+
+  CURRENT_DATE=`date +%l:%M%p`
+  read -r -p "Does this deploy include a migration that need to run?  Quill policy is to not run migrations between the hours of 10am and 3pm Eastern time on school days unless it is an emergency.  Your local time is $CURRENT_DATE.  You may cancel your deploy with ctrl-c.  If you want to continue this deploy, press any key."  response
+  sh ../../scripts/post_slack_deploy.sh $app_name prod "$HEROKU_APP promotable slug" false
+  sh ../../scripts/post_slack_deploy_description.sh $app_name
+}
+
+for arg in "$@"; do
+  case $arg in
+    --osx-notification)
+      osx_notification=1
+      ;;
+  esac
+  case "$arg" in
+    -p|--promote)
+      promote
```

Comment:
The `--promote` flag intercepts the normal flow, then exits, so the normal deploy flow is untouched.